pylint/config/arguments_manager.py

```diff
@@ -43,25 +43,34 @@ def __init__(self) -> None:
 
     def _register_options_provider(self, provider: "checkers.BaseChecker") -> None:
         """Register an options provider and load its defaults."""
-        # pylint: disable-next=fixme
-        # TODO: Do something own_group parameter (see OptionsManagerMixIn.register_options_provider)
         for opt, optdict in provider.options:
             argument = _convert_option_to_argument(opt, optdict)
-            self._add_arguments_to_parser(provider.name, argument)
+            section = argument.section or provider.name.capitalize()
 
-        # pylint: disable-next=fixme
-        # TODO: Do something with option groups within optdicts (see OptionsManagerMixIn.register_options_provider)
+            # pylint: disable-next=fixme
+            # TODO: Optparse: Always define option_groups_descs on ArgumentsProvider
```

Comment:
`ArgumentsProvider` hasn't been merged yet. But this should be an attribute on it, not on `BaseChecker` (or on `PyLinter` as it currently is...)